pkgs/misc/nflz/default.nix

```diff
@@ -0,0 +1,34 @@
+{ fetchCrate
+, lib
+, rustPlatform
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "nflz";
+  version = "1.0.2";
+
+  src = fetchCrate {
+    inherit pname version;
+    sha256 = "sha256-c9+79zrIU/M1Rh+DiaLJzbrNSa4IKrYk1gP0dsabUiw=";
```

Comment:
```suggestion
    hash = "sha256-c9+79zrIU/M1Rh+DiaLJzbrNSa4IKrYk1gP0dsabUiw=";
```

Reply:
My other packages (2 weeks ago or so) were accepted with `sha256=` and `cargoSha256=`. Can you please explain who is right? From a quick glance, both are used in nixpkgs for existing rust packages.

Reply:
Under the code they are mostly interchangeable, or more precisely there is a legacy compatibility layer.

My intention here is to be future-proof.

Not a long time ago we employed MD5 hashes, but nowadays we use SHA256 (and there is some support for SHA512).

When in the future a new super duper hash algorithm arrives, it will be better to write, say,

`hash = "sha1024-xxxxxxxx";`

instead of

`sha1024 = "sha1024-xxxxxxxx";`

(Indeed, NetBSD is already using a new hash algo, Blake2. This is not a too distant future, after all...)